src/lvb.py

```diff
@@ -0,0 +1,21 @@
+# Written by David Bunn, 2024 (modified from mpboot.py)
+# Generates LVB command for Maximum Parsimony stage
+import re
+
+def GenerateLVBCommand(initial_software: str, msa_path: str) -> str:
+    software_path = f'{initial_software} '
+    pass_msa_path = "-i " + msa_path
+
+    file_extension = re.search(".(\w+)\Z", msa_path)
+    formats = {'phy':'phylip','ph':'phylip','phylip':'phylip',
```

Comment:
i'm not sure you need the 'phylip' extensions here, as you have the else on line 17, which assigns to phylip anyway

Reply:
Since there's no requirement for users to name their files sensibly, I suggest giving this a stress-test - a Fasta-format file ending in ".aln", a Nexus-format file ending in ".nex" etc. We would hope for a graceful crash in these cases - not, for example, "core dumped". These tests could be scripted and added to the test suite.

Reply:
Hi Joe, I included the phylip extensions so that if these were used in the input file, it would print 'PHYLIP format detected'. If you had an unknown extension, phylip would be assumed but this wouldn't be printed (in this case it hasn't detected any format). Do you see what I mean? Happy to change if you'd prefer.
Daniel, this certainly sounds like a good idea. I might hold off for now and discuss this with you on Wednesday as I'm not sure how to go about this in practice.